docker/docker_client.go

```diff
@@ -998,20 +998,33 @@ func (c *dockerClient) getExternalBlob(ctx context.Context, urls []string) (io.R
 			resp.Body.Close()
 			continue
 		}
-		return resp.Body, getBlobSize(resp), nil
+
+		size, err := getBlobSize(resp)
+		if err != nil {
+			size = -1
```

Comment:
Should we at least logrus.Debug the error?

Reply:
In both cases, `GetBlob` is explicitly allowed to return -1, so we wouldn’t expect users to need to diagnose why this is happening. Typically, the size is only used to sanity-check sizes of uploads, but the digest verification should serve the same purpose anyway.

The only adverse effect I could find is that writes to `docker-daemon` or `docker-archive` might need to store a file to disk if the size is unknown, _and_ the input is uncompressed (if the input is compressed, the transports need to store it to disk in either case).

I don’t think this is really worth worrying about, and probably not worth the log noise, but I don’t feel strongly about this at all.